avae/data.py

```diff
@@ -0,0 +1,240 @@
+            shuffle=True,
+            drop_last=True,
+        )
+        vals = DataLoader(
+            val_data,
+            batch_size=batch_s,
+            num_workers=0,
+            shuffle=True,
+            drop_last=(not no_val_drop),
+        )
+        tests = []
+        if len(vals) < 1 or len(trains) < 1:
+            # ensure the batch size is not smaller than validation set
+            raise RuntimeError(
+                "Validation or train set is too small for the current batch "
+                "size. Please edit either split percent '-sp/--split' or batch"
+                " size '-ba/--batch' or set '-nd/--no_val_drop flag' (only if "
+                "val is too small). Batch: {}, train: {}, val: {}, "
+                "split: {}%.".format(
+                    batch_s, len(train_data), len(val_data), splt
+                )
+            )
+        print("Train / val batches:", len(trains), len(vals))
+        print()
+
+        if affinity is not None:
+            lookup = lookup.to_numpy(dtype=np.float32)
+        else:
+            lookup = None
+
+    if eval or ("test" in os.listdir(datapath)):
+        if "test" in os.listdir(datapath):
+            datapath = os.path.join(datapath, "test")
+        data = ProteinDataset(datapath, lim=lim, collect_m=collect_meta)
+        print("Eval data size:", len(data))
+        tests = DataLoader(
+            data, batch_size=batch_s, num_workers=0, shuffle=True
+        )
+        print("Eval batches:", len(tests))
+        print()
+
+    if eval:
+        return tests
+    else:
+        return trains, vals, tests, lookup  # , dsize
+
+
+class ProteinDataset(Dataset):
+    """Protein dataset. Opens MRC files and returns images along with their
+    affinity and associated metadata.
+
+    Parameters
+    ----------
+    root_dir : string
+        Base directory containing .mrc files.
+    amatrix : pd.DataFrame
+        A square symmetric matrix where each column and row is the index of an
+        object class from the training set,
+        consisting of M different classes. First row and column contain IDs of
+        the classes.
+    transform: torchvision.transforms.Transform
+        List of transforms to be applied to the images.
+    lim : int
+        Limit the dataset size to the given number; useful for debugging
+        purposes.
+    """
+
+    def __init__(
+        self,
+        root_dir,
+        amatrix=None,
+        classes=None,
+        transform=None,
+        lim=None,
+        collect_m=False,
+    ):
+        super().__init__()
+
+        self.collect_meta = collect_m
+
+        self.amatrix = amatrix
+
+        self.root_dir = root_dir
+        self.paths = [f for f in os.listdir(root_dir) if ".mrc" in f]
+        random.shuffle(self.paths)
+        ids = np.unique([f.split("_")[0] for f in self.paths])
+        self.final_classes = ids
+
+        if classes is not None:
+            classes_list = pd.read_csv(classes).columns.tolist()
```

Comment:
This only returns the first row of the list, we could go back to the last way of reading this file:

```suggestion
with open(classes, newline="\n") as f:
    classes_list = np.asarray(f.read().splitlines()).flatten()
```

Reply:
decided to run with comma separated list